lib/rucio/core/did_meta_plugins/elasticsearch_meta.py

```diff
@@ -0,0 +1,406 @@
+    def drop_index(self) -> None:
+        self.client.indices.delete(index=self.index)
+
+    def get_metadata(
+        self,
+        scope: "InternalScope",
+        name: str,
+        *,
+        session: "Optional[Session]" = None
+    ) -> dict[str, Any]:
+        """
+        Get data identifier metadata.
+
+        :param scope: The scope name
+        :param name: The data identifier name
+        :param session: The database session in use
+        :returns: The metadata for the did
+        :raises DataIdentifierNotFound: If the DID metadata is not found.
+        :raises RucioException: If another error occurs during the process.
+        """
+
+        doc_id = f"{scope.internal}{name}"
+        try:
+            doc = self.client.get(index=self.index, id=doc_id)["_source"]
+        except elastic_exceptions.NotFoundError as err:
+            raise exception.DataIdentifierNotFound(f"No metadata found for DID '{scope}:{name}' not found") from err
+        except Exception as err:
+            raise exception.RucioException(err)
+        return doc
+
+    def set_metadata(
+        self,
+        scope: "InternalScope",
+        name: str,
+        key: str,
+        value: str,
+        recursive: bool = False,
+        *,
+        session: "Optional[Session]" = None
+    ) -> None:
+        """
+        Set single metadata key.
+
+        :param scope: the scope of did
+        :param name: the name of the did
+        :param key: the key to be added
+        :param value: the value of the key to be added
+        :param recursive: recurse into DIDs (not supported)
+        :param session: The database session in use
+        :raises DataIdentifierNotFound: If the DID is not found.
+        :raises RucioException: If an error occurs while setting the metadata.
+        """
+        self.set_metadata_bulk(scope=scope, name=name, meta={key: value}, recursive=recursive, session=session)
+
+    def set_metadata_bulk(
+        self,
+        scope: "InternalScope",
+        name: str,
+        meta: dict[str, Any],
+        recursive: bool = False,
+        *,
+        session: "Optional[Session]" = None
+    ) -> None:
+        """
+        Bulk set metadata keys.
+
+        :param scope: the scope of did
+        :param name: the name of the did
+        :param meta: dictionary of metadata keypairs to be added
+        :param recursive: recurse into DIDs (not supported)
+        :param session: The database session in use
+        :raises DataIdentifierNotFound: If the DID is not found.
+        :raises RucioException: If an error occurs while setting the metadata.
+        """
+        doc_id = f"{scope.internal}{name}"
+        try:
+            # Try to get existing metadata
+            existing_meta = self.get_metadata(scope, name)
+        except exception.DataIdentifierNotFound:
+            existing_meta = {
+                'scope': str(scope.external),
+                'name': str(name),
+                'vo': str(scope.vo)
```

Comment:
```suggestion
                'vo': scope.vo
```

`InternalScope.vo` is always a `str`